atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -0,0 +1,157 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from atomic_reactor.plugin import PreBuildPlugin
+
+import codecs
+import json
+import jsonschema
+from pkg_resources import resource_stream
+import yaml
+
+
+# Key used to store the config object in the plugin workspace
+WORKSPACE_CONF_KEY = 'reactor_config'
+
+
+def get_config(workflow):
+    """
+    Obtain configuration object
+    Does not fail
+
+    :return: ReactorConfig instance
+    """
+    try:
+        workspace = workflow.plugin_workspace[ReactorConfigPlugin.key]
+        return workspace[WORKSPACE_CONF_KEY]
+    except KeyError:
+        # The plugin did not run or was not successful: use defaults
+        conf = ReactorConfig()
+        workspace = workflow.plugin_workspace.get(ReactorConfigPlugin.key, {})
+        workspace[WORKSPACE_CONF_KEY] = conf
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = workspace
+        return conf
+
+
+class ClusterConfig(object):
+    """
+    Configuration relating to a particular cluster
+    """
+
+    def __init__(self, name, max_concurrent_builds, enabled=True):
+        self.name = str(name)
+        self.max_concurrent_builds = int(max_concurrent_builds)
+        self.enabled = enabled
+
+
+class ReactorConfig(object):
+    """
+    Class to parse the atomic-reactor configuration file
+    """
+
+    VERSION_KEY = 'version'
+    CLUSTERS_KEY = 'clusters'
+
+    DEFAULT_CONFIG = {'version': 1}
+
+    def __init__(self, config=None):
+        self.conf = config or self.DEFAULT_CONFIG
+
+        version = self.conf[self.VERSION_KEY]
+        if version != 1:
```

Comment:
ahh, I was thinking that DEFAULT_CONFIG might used as well asthe "latest" version allowed